src/mantine-hooks/src/use-valid-state/use-valid-state.ts

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+
+export function useValidState<T>(
```

Comment:
useValidState hook:
- rename to useValidatedState
- initialValue should be first argument, it should be required
- remove return type definition, ts handles that automatically `readonly [validValue: T, setValue: (value: T) => void, value: T]`
- Hook must return current validation state and thus have this definition:
```tsx
const [{ value, lastValidValue, valid }, onChange] = useValidatedState('hello', (val) => val === 'hello');
```